GAWB-293 add Intersection Groups to Workspace

jmthibault79 opened the pull request:
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell ![](http://i.imgur.com/9dLzbPd.png) that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] ![](http://i.imgur.com/9dLzbPd.png) sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Ask QA if it's a convenient time to merge this PR**; hold off if they're in the middle of testing something
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

jmthibault79 committed:
GAWB-293 add Intersection Groups to Workspace

dvoet commented:
deleteWorkspace in WorkspaceService needs to delete the intersection groups and while you are at it AuthDAO.deleteWorkspaceAccessGroups should be in the WorkspaceService not in the DAO. If you would rather do this in another PR, fine, but enter a jira.

dvoet commented on src/test/scala/org/broadinstitute/dsde/rawls/integrationtest/HttpGoogleServicesDAOSpec.scala:
```diff
@@ -54,16 +54,16 @@ class HttpGoogleServicesDAOSpec extends FlatSpec with Matchers with IntegrationT
 
   "HttpGoogleServicesDAO" should "do all of the things" in {
-    val googleWorkspaceInfo = Await.result(gcsDAO.setupWorkspace(testCreator, testProject, testWorkspaceId, testWorkspace), Duration.Inf)
+    val googleWorkspaceInfo = Await.result(gcsDAO.setupWorkspace(testCreator, testProject, testWorkspaceId, testWorkspace, None), Duration.Inf)
```
add to this test to make sure your new code works

dvoet commented:
@jmthibault79 done with review

jmthibault79 commented:
Addressed comments, fixed a bug in the rollback code, and updated getWorkspaceStatus, which I had missed before.

Cleaning up DAOs is in https://broadinstitute.atlassian.net/browse/GAWB-311

dvoet commented:
:+1:
